File: frontend/novera-frontend/src/service/paymentService.ts
import axiosInstance from "../api/axios";

import type {

    CreatePaymentRequest,

    CreatePaymentResponse,

    VerifyPaymentRequest,

    VerifyPaymentResponse,

} from "../types/payment";

const PAYMENT_BASE_URL =
    `${import.meta.env.VITE_PAYMENT_API}/api/payments`;

const createPayment = async (
    request: CreatePaymentRequest
): Promise<CreatePaymentResponse> => {

    const { data } =
        await axiosInstance.post(
            PAYMENT_BASE_URL,
            request
        );

    return data;
};

const verifyPayment = async (
    request: VerifyPaymentRequest
): Promise<VerifyPaymentResponse> => {

    const { data } =
        await axiosInstance.post(
            `${PAYMENT_BASE_URL}/verify`,
            request
        );

    return data;
};

export default {

    createPayment,

    verifyPayment,

};

File: paymnet-service/src/main/java/com/example/novera/service/PaymentService.java
package com.example.novera.service;

import com.example.novera.dto.CreatePaymentRequest;
import com.example.novera.dto.CreatePaymentResponse;
import com.example.novera.dto.PaymentOrderResponse;
import com.example.novera.dto.VerifyPaymentRequest;
import com.example.novera.dto.VerifyPaymentResponse;


public interface PaymentService {

    CreatePaymentResponse createPayment(CreatePaymentRequest request,
                                        Long userId);

    VerifyPaymentResponse verifyPayment(
        VerifyPaymentRequest request,
        Long userId);

}
